plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/action/NewJavaSourceFileAction.java

```diff
@@ -57,22 +61,20 @@ public void actionPerformed(ActionEvent e) {
 
     @Override
     public void updateProjectAction(ActionEvent e) {
-        Selection<?> selection = projectExplorer.getSelection();
-
-        if (selection == null) {
+        CurrentProject project = appContext.getCurrentProject();
+        if (project == null || !"maven".equals(project.getRootProject().getType())) {
```

Comment:
We have special constant for Maven PType ID: org.eclipse.che.ide.extension.maven.shared.MavenAttributes#MAVEN_ID